internal/binding/local.go

```diff
@@ -188,20 +437,44 @@ func (p *chanPort) ReadPacketData() ([]byte, gopacket.CaptureInfo, error) {
 	if p.rxQueue == nil {
 		return nil, gopacket.CaptureInfo{}, nil
 	}
-	data := (<-p.rxQueue.Receive()).([]byte)
-	return data, gopacket.CaptureInfo{
+	data, ok := <-p.rxQueue.Receive()
+	if !ok {
+		return nil, gopacket.CaptureInfo{}, io.EOF
+	}
+	packet := data.([]byte)
+	return packet, gopacket.CaptureInfo{
 		Timestamp:     time.Now(),
-		CaptureLength: len(data),
-		Length:        len(data),
+		CaptureLength: len(packet),
+		Length:        len(packet),
 	}, nil
 }
 
+func (p *chanPort) Close() {
+	p.txQueue.Close()
+}
+
+func (p *chanPort) LinkType() layers.LinkType {
+	return layers.LinkTypeEthernet
+}
+
+func (p *chanPort) SetBPFFilter(string) error {
+	return nil
+}
+
 type portMgr struct {
-	ports           map[string]*chanPort
+	dut *dutPortMgr
+	ate *atePortMgr
+}
+
+// dutPortMgr implements lemming's port creator interface.
+// Note: this interface is different from magna, to avoid making the two project depending on eachother.
+type dutPortMgr struct {
+	ports map[string]*chanPort
+	// TODO: lastCreatedPort is a hack because we don't currently model hardware ports correctly in lemming.
 	lastCreatedPort *chanPort
 }
 
-func (lm *portMgr) CreatePort(name string) (fwdcontext.Port, error) {
+func (lm *dutPortMgr) CreatePort(name string) (fwdcontext.Port, error) {
```

Comment:
Should we use a different alias than `lm`?

Reply:
changed to mgr